trie-db/test/src/triedbmut.rs

```diff
@@ -881,12 +848,205 @@ fn test_two_assets_memory_db_inner_2<T: TrieLayout>() {
 	assert_eq!(state.get(key1.as_ref()).unwrap().unwrap(), data1);
 	assert_eq!(state.get(key2.as_ref()).unwrap().unwrap(), data2);
 
-	state.commit();
-
 	let key3 = [1u8, 3];
 	let data3 = [3u8; 2];
 	state.insert(key3.as_ref(), &data3).unwrap();
 	assert_eq!(state.get(key1.as_ref()).unwrap().unwrap(), data1);
 	assert_eq!(state.get(key2.as_ref()).unwrap().unwrap(), data2);
 	assert_eq!(state.get(key3.as_ref()).unwrap().unwrap(), data3);
 }
+
+test_layouts!(child_trie, child_trie_internal);
+fn child_trie_internal<T: TrieLayout, DB: TestDB<T>>() {
```

Comment:
changes from arkpar/new-trie important to review as gives a good view on usage for substrate (see also https://github.com/cheme/polkadot-sdk/tree/cheme/new-trie where I find the use of iterator a bit awkward when calculating root but a given struct encapsulating it may make things ok).

Reply:
Loogs good, apart from the naming.